### PR TITLE
Refactor SkyWalking.Diagnostics

### PR DESCRIPTION
**Pull request by liuhaoyang**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues
___
### New feature or improvement
- Describe the details and related test reports.

Remove Microsoft.Extensions.DiagnosticsAdapter

### CODE REVIEW

File: src/SkyWalking.AspNetCore/SkyWalking.AspNetCore.csproj [hunk 1]
```diff
@@ -15,7 +15,6 @@
     <PackageReference Include="Microsoft.AspNetCore.Hosting" Version="2.0.0" />
     <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" Version="2.0.0" />
     <PackageReference Include="Microsoft.AspNetCore.Mvc.Abstractions" Version="2.0.0" />
-    <PackageReference Include="Microsoft.Extensions.DiagnosticAdapter" Version="2.1.0-preview2-final" />
     <PackageReference Include="Microsoft.Extensions.Http" Version="2.1.0-preview2-final" />
```
Comment (yuleyule66) [inline src/SkyWalking.AspNetCore/SkyWalking.AspNetCore.csproj]: can we changes the package version to 2.0.0 ?

Comment (liuhaoyang) [reply src/SkyWalking.AspNetCore/SkyWalking.AspNetCore.csproj]: `Microsoft.Extensions.DiagnosticAdapter` should be removed.

Comment (yuleyule66) [reply src/SkyWalking.AspNetCore/SkyWalking.AspNetCore.csproj]: package `Microsoft.Extensions.Http` and `Microsoft.Extensions.Logging.Abstractions` and `Microsoft.Extensions.Options` versions , Maybe it's not part of this submission.

Comment (liuhaoyang) [reply src/SkyWalking.AspNetCore/SkyWalking.AspNetCore.csproj]: Refactoring HttpClientFactory should be in a new PR.

Comment (liuhaoyang) [reply src/SkyWalking.AspNetCore/SkyWalking.AspNetCore.csproj]: These submissions are only refactoring `SkyWalking.Diagnostics`.

Comment (yuleyule66) [reply src/SkyWalking.AspNetCore/SkyWalking.AspNetCore.csproj]: Agree !

**Pull request merged**